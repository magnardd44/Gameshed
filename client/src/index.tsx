import * as React from 'react';
import { Component } from 'react-simplified';
import ReactDOM from 'react-dom';
import {
  Card,
  Alert,
  Row,
  Form,
  Column,
  Button,
  NavBar,
  Header,
  Container,
  ColumnCentre,
  SearchResult,
} from './widgets';
import axios from 'axios';
import { HashRouter, Route } from 'react-router-dom';
import { AddReview, PublishReview } from './review-components';
import { Hash } from 'crypto';
import { createHashHistory } from 'history';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

const history = createHashHistory();

class NavHeader extends Component {
  render() {
    return (
      <div>
        <NavBar brand="GameShed">
          <NavBar.Link to="/games">Games</NavBar.Link>
        </NavBar>
        <Header></Header>
      </div>
    );
  }
}

class Search extends Component {
  input = '';
  render() {
    return (
      <>
        <Container textalign={'centre'}>
          <Row>
            <ColumnCentre width={6} offset={2}>
              <Form.Input
                type={this.input}
                value={this.input}
                placeholder="Søk etter et spill"
                onChange={(event) => (this.input = event.currentTarget.value)}
              />
            </ColumnCentre>
            <ColumnCentre width={2}>
              <Button.Success
                onClick={() => {
                  this.search();
                }}
              >
                Søk
              </Button.Success>
            </ColumnCentre>
          </Row>
        </Container>
      </>
    );
  }
  search() {
    console.log('search');
    history.push('/results');
  }
}

class SearchListings extends Component {
  render() {
    return (
      <Container>
        Søkeresultater:
        <SearchResult></SearchResult>
        <SearchResult></SearchResult>
        <SearchResult></SearchResult>
      </Container>
    );
  }
}

ReactDOM.render(
  <HashRouter>
    <div>
      <NavHeader />
      <Route exact path="/" component={Search}></Route>
      <Route exact path="/results" component={SearchListings}></Route>
      <Route exact path="/addReview" component={AddReview} />
      <Route exact path="/publishReview/:id" component={PublishReview} />
    </div>
  </HashRouter>,
  document.getElementById('root')
);

/*
class Start extends Component {
  input = '';
  stdout = '';
  stderr = '';
  errCode: number | null = null;
}



  render() {
    return (
      <>
        <Card title="GameShed">
          <Row>
            <Column>
              <Form.Label>Søk:</Form.Label>
              <Form.Textarea
                value={this.input}
                onChange={(event) => (this.input = event.currentTarget.value)}
              />
            </Column>
          </Row>
          <Button.Success
            onClick={() => {
              axios
                .post<{ exitStatus: number; stdout: string; stderr: string }>('/run', {
                  language: 'js',
                  source: this.input,
                })
                .then((response) => {
                  this.errCode = response.data.exitStatus;
                  this.stdout = response.data.stdout;
                  this.stderr = response.data.stderr;
                })
                .catch((error: Error) => Alert.danger('Could not run app.js: ' + error.message));
            }}
          >
            Søk etter spill
          </Button.Success>
        </Card>
        <Card title="Standard output">{this.stdout}</Card>
        <Card title="Standard error">{this.stderr}</Card>
        <Card title="Errorcode">{this.errCode}</Card>
      </>
    );
  }
}


const root = document.getElementById('root');
if (root)
  ReactDOM.render(
    <HashRouter>
      <Alert />

      <Route exact path="/" component={Start} />
      
    </HashRouter>,
    root
  );
 
 */
