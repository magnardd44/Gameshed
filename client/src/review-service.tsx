import axios from 'axios';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Review = {
  id: number;
  title: string;
  text: boolean;
  rating: number;
};

class ReviewService {
  /**
   * Get task with given id.
   */
  get(id: number) {
    return axios.get<Review>('/reviews/' + id).then((response) => response.data);
  }

  /**
   * Get all tasks.
   */
  getAll() {
    return axios.get<Review[]>('/reviews').then((response) => response.data);
  }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  create(title: string, text: string, rating: number) {
    return axios
      .post<{ id: number }>('/reviews', { review_title: title, text: text, rating: rating })
      .then((response) => response.data.id);
  }
}

const reviewService = new ReviewService();
export default reviewService;
