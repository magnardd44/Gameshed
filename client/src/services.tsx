import axios from 'axios';
import { sharedComponentData } from 'react-simplified';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Game = {
  game_id: number;
  game_title: string;
  genre: string[];
  genre_id: number;
  platform: string[];
  game_description: string;
};

export type Review = {
  review_id: number;
  game_id: string;
  external_game_id: string[];
  game_title: number;
  genre: string[];
  platform: string;
  review_title: string;
  text: string;
  user_id: number;
  rating: number;
  published: boolean;
};

class GameService {
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: [],
    genre_id: 0,
    platform: [],
    game_description: '',
  };
  games: Game[] = [];

  /**
   * Get game with given id.
   */
  get(id: number) {
    return axios.get<Game>('/games/' + id).then((response) => response.data);
  }

  /**
   * Get all games.
   */
  getAll() {
    return axios.get<Game[]>('/games').then((response) => response.data);
  }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  create(title: string, genre: string, platform: number, description: string) {
    return axios
      .post<{ id: number }>('/games', {
        game_title: title,
        genre: genre,
        platform: platform,
        game_description: description,
      })
      .then((response) => response.data.id);
  }
}

class ReviewService {
  review: Review = {
    review_id: 0,
    game_id: '',
    external_game_id: [],
    game_title: 0,
    genre: [],
    platform: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
  };
  reviews: Review[] = [];

  /**
   * Get review with given id.
   */
  get(id: number) {
    return axios.get<Review>('/reviews/' + id).then((response) => response.data);
  }

  /**
   * Get all reviews.
   */
  getAll() {
    return axios.get<Review[]>('/reviews').then((response) => response.data);
  }

  /**
   * Create new review having the given title.
   *
   * Resolves the newly created review id.
   */
  create(title: string, text: string, rating: number) {
    return axios
      .post<{ id: number }>('/reviews', { review_title: title, text: text, rating: rating })
      .then((response) => response.data.id);
  }
}

export let gameService = sharedComponentData(new GameService());
export let reviewService = sharedComponentData(new ReviewService());
