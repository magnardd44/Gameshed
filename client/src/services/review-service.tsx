import axios from 'axios';
import { sharedComponentData } from 'react-simplified';
import userService from './user-service';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Oauth_token = {
  access_token: string;
  expire_time: number;
  token_type: string;
};

class SearchService {
  token: Oauth_token = {
    access_token: '',
    expire_time: 0,
    token_type: '',
  };
}

export type Review = {
  review_id: number;
  game_id: number;
  game_title: string;
  review_title: string;
  text: string;
  user_id: number;
  rating: number;
  published: boolean;
  genre_id: number;
  platform_id: number;
  relevant: number;
  likes: number;
};

export type RelatedReview = {
  review_id: number;
  game_id: number;
  game_title: string;
  review_title: string;
  text: string;
  user_id: number;
  rating: number;
  published: boolean;
  genre_id: number;
  platform_id: number;
  relevant: number;
  likes: number;
};

class ReviewService {
  review: Review = {
    review_id: 0,
    game_id: 0,
    game_title: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
    genre_id: 0,
    platform_id: 0,
    relevant: 0,
    likes: 0,
  };
  reviews: Review[] = [];

  relatedReview: RelatedReview = {
    review_id: 0,
    game_id: 0,
    game_title: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
    genre_id: 0,
    platform_id: 0,
    relevant: 0,
    likes: 0,
  };

  relatedReviews: RelatedReview[] = [];

  publishedreviews: Review[] = [];

  drafts: Review[] = [];

  /**
   * Get review with given id.
   */
  get(id: number) {
    return userService.axios.get<Review>('/reviews/review/' + id).then((response) => response.data);
  }

  /**
   * Get review with given id.
   */
  getAllByGameId(id: number) {
    return userService.axios.get<Review>('/reviews/' + id).then((response) => response.data);
  }

  /**
   * Get reviews with given id.
   */
  getAllById(user_id: number) {
    return axios.get<Review[]>('/reviews/users/' + user_id).then((response) => response.data);
  }

  /**
   * Get reviews with given id.
   */
  getAll() {
    return axios.get<Review[]>('/reviews/').then((response) => response.data);
  }

  /**
   * Get published reviews
   */
  getPublishedReviews() {
    return axios.get<Review[]>('/reviews/published').then((response) => response.data);
  }

  /**
   * Get reviews based on genre
   */
  getGenre(genre_id: number) {
    return axios.get<Review[]>('/reviews/genre/' + genre_id).then((response) => response.data);
  }

  //Get reviews based on platform
  getPlatform(platform_id: number) {
    return axios
      .get<Review[]>('/reviews/platform/' + platform_id)
      .then((response) => response.data);
  }

  /**
   * Create new review   *
   * Resolves the newly created review id.
   */
  create(game_id: number, title: string, text: string, rating: number) {
    return userService.axios
      .post<{ id: number }>('/reviews', {
        game_id: game_id,
        review_title: title,
        text: text,
        rating: rating,
      })
      .then((response) => response.data.id);
  }

  //Change review status to published
  publish(id: number) {
    return userService.axios
      .patch<Review>('/reviews/' + id + '/publish')
      .then((response) => response.data.review_id);
  }
  //Add like to review
  like(review_id: number, user_id: number, relevant: number) {
    return userService.axios
      .patch<Review>('/reviews/' + review_id + '/relevant', {
        relevant: relevant,
        user_id: user_id,
      })
      .then((response) => response.data);
  }

  //Edit review before publishing
  edit(id: number, title: string, text: string, rating: number) {
    return userService.axios
      .patch<Review>('/reviews/' + id, { title: title, text: text, rating: rating })
      .then((response) => response.data);
  }

  //Delete review
  delete(id: number) {
    return userService.axios
      .delete<Review>('reviews/' + id)
      .then((response) => response.data.review_id);
  }
}

export let reviewService = sharedComponentData(new ReviewService());
