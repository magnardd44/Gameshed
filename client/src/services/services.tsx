import axios from 'axios';
import { sharedComponentData } from 'react-simplified';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Game = {
  game_id: number;
  game_title: string;
  genre: string[];
  genre_id: number;
  platform: string[];
  game_description: string;
};

class GameService {
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: [],
    genre_id: 0,
    platform: [],
    game_description: '',
  };
  games: Game[] = [];

  /**
   * Get game with given id.
   */
  get(id: number) {
    return axios.get<Game>('/games/' + id).then((response) => response.data);
  }

  /**
   * Get all games.
   */
  getAll() {
    return axios.get<Game[]>('/games').then((response) => response.data);
  }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  create(title: string, genre: string, platform: number, description: string) {
    return axios
      .post<{ id: number }>('/games', {
        game_title: title,
        genre: genre,
        platform: platform,
        game_description: description,
      })
      .then((response) => response.data.id);
  }
}

export let gameService = sharedComponentData(new GameService());
