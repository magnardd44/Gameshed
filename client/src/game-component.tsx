import * as React from 'react';
import { Component } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { gameService, reviewService, Game, Review } from './services';
import { createHashHistory } from 'history';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student

export class GameCard extends Component {
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: [],
    genre_id: 0,
    platform: [],
    game_description: '',
  };
  description =
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco...';
  render() {
    return (
      <Card title={this.game.game_title}>
        <h6 className="card-subtitle mb-2 text-muted">
          Terningkast:
          <ThumbNail small img="https://cdn-icons-png.flaticon.com/512/220/220725.png"></ThumbNail>
        </h6>
        <Row>
          <Column width={2}>
            <ThumbNail img="https://cdn-icons-png.flaticon.com/512/686/686589.png"></ThumbNail>
          </Column>
          <Column width={6}>
            {this.game.game_description}
            <Linebreak></Linebreak>
          </Column>
          <Column width={2}></Column>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <Column>Sjanger: {this.game.genre}</Column>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <Column>Platform: {this.game.platform}</Column>
        </Row>
      </Card>
    );
  }
  /** return axios
   */
  mounted() {
    gameService.getAll().then((result) => {
      this.game = result[0];
      console.log(this.game);
    });
  }
}
