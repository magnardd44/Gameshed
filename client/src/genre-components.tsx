import * as React from 'react';
import { Component, sharedComponentData } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
  FormContainer,
  FormGroup,
  CategoryCard,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { gameService, Game } from './services/game-services';
import { Genre } from './services/genre-service';
import { createHashHistory } from 'history';
import axios from 'axios';
import { genreService } from './services/genre-service';
import { Review, reviewService } from './services/review-service';

const history = createHashHistory();

export class Category extends Component {
  genres: Genre[] = [];
  reviews: Review[] = [];
  games: Game[] = [];
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: 0,
    genre_id: 0,
    platform: 0,
    game_description: '',
    igdb_id: 0,
    genres: [],
    platforms: [],
  };
  review: Review = {
    review_id: 0,
    game_id: 0,
    game_title: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
    genre_id: 0,
    relevant: 0,
    platform_id: 0,
  };
  state = { isHidden: false };
  genreCall(id: number) {
    reviewService
      .getGenre(id)
      .then((data) => {
        console.log(data);
        this.reviews = data;
      })
      .catch((error) => Alert.danger('Error retrieving reviews: ' + error.message));
    this.setState({ isHidden: false });
  }
  render() {
    return (
      <Container>
        Velg kategori/sjanger:
        <Row>
          <ColumnCentre width={12} smwidth={4} mdwidth={2}>
            {this.genres.map((genre) => (
              <CategoryCard
                key={genre.genre_id}
                img={
                  genre.genre_img
                    ? 'https://helenaagustsson.github.io/INFT2002-images/images/' +
                      genre.genre_img +
                      '.png'
                    : 'https://helenaagustsson.github.io/INFT2002-images/images/' +
                      genre.genre_name +
                      '.png'
                }
              >
                <Button.Success onClick={() => this.genreCall(genre.genre_id)}>
                  {genre.genre_name}
                </Button.Success>
              </CategoryCard>
            ))}
          </ColumnCentre>
          <ColumnCentre>
            {this.state.isHidden ? null : (
              <div className="sticky-top">
                {this.reviews.map((review, index) => (
                  <Row key={index}>
                    <Column>{review.game_title}</Column>
                    <Column>
                      <NavLink to={'/publishedReviews/' + review.review_id}>
                        {review.review_title}
                      </NavLink>
                    </Column>
                    <Column>{review.rating}</Column>
                  </Row>
                ))}
              </div>
            )}
          </ColumnCentre>
        </Row>
      </Container>
    );
  }
  mounted() {
    genreService
      .getAll()
      .then((genres) => {
        this.genres = genres;
      })
      .catch((error) => Alert.danger('Error getting reviews: ' + error.message));
  }
}
