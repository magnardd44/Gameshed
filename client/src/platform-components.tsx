import * as React from 'react';
import { Component, sharedComponentData } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
  FormContainer,
  FormGroup,
  CategoryCard,
  ReviewCard,
  DropDown,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { gameService, Game } from './services/game-services';
import { Genre } from './services/genre-service';
import { createHashHistory } from 'history';
import axios from 'axios';
import { genreService } from './services/genre-service';
import { platformService } from './services/platform-service';
import { Review, reviewService } from './services/review-service';
import { RelatedReviews } from './related-reviews';

const history = createHashHistory();

//Renders overview of published reviews based on genre
export class Platform extends Component {
  reviews: Review[] = [];
  games: Game[] = [];
  state = { isHidden: true };

  review: Review = {
    review_id: 0,
    game_id: 0,
    game_title: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
    genre_id: 0,
    relevant: 0,
    platform_id: 0,
    likes: 0,
  };

  platformCall(id: number) {
    reviewService
      .getPlatform(id)
      .then((data) => {
        this.setState({ isHidden: false });
        console.log(data);
        this.reviews = data;
      })
      .catch((error) => Alert.danger('Error retrieving reviews: ' + error.message));
  }

  render() {
    return (
      <>
        <Container>
          <h3 className="text-center">Her kan du søke anmeldelser etter platform</h3>
          <div className="text-center">
            <Button.Success
              small
              onClick={() => {
                history.push('/reviews-by-genre');
              }}
            >
              Klikk her
            </Button.Success>{' '}
            for å søke etter sjanger
          </div>
          <Linebreak></Linebreak>
          <h5>Velg platform:</h5>
          <Row>
            <ColumnCentre width={6} smwidth={4} mdwidth={2}>
              <Card title="Play Station 4">
                <Button.Success onClick={() => this.platformCall(145)}>Vis</Button.Success>
              </Card>
              <Card title="Play Station 5">
                <Button.Success onClick={() => this.platformCall(146)}>Vis</Button.Success>
              </Card>
              <Card title="Nintendo Switch">
                <Button.Success onClick={() => this.platformCall(119)}>Vis</Button.Success>
              </Card>
              <Card title="XBox One">
                <Button.Success onClick={() => this.platformCall(186)}>Vis</Button.Success>
              </Card>
              <Card title="XBox 360">
                <Button.Success onClick={() => this.platformCall(185)}>Vis</Button.Success>
              </Card>
              <Card title="PC - Microsoft">
                <Button.Success onClick={() => this.platformCall(126)}>Vis</Button.Success>
              </Card>
              <Card title="Gameboy">
                <Button.Success onClick={() => this.platformCall(89)}>Vis</Button.Success>
              </Card>
              <Card title="Mac">
                <Button.Success onClick={() => this.platformCall(100)}>Vis</Button.Success>
              </Card>
              <Card title="IOS">
                <Button.Success onClick={() => this.platformCall(98)}>Vis</Button.Success>
              </Card>
              <Card title="Nintendo DS">
                <Button.Success onClick={() => this.platformCall(114)}>Vis</Button.Success>
              </Card>
              <Card title="Andre">
                <Button.Success onClick={() => this.platformCall(101 + 102)}>Vis</Button.Success>
              </Card>
            </ColumnCentre>
            <ColumnCentre>
              {' '}
              {this.state.isHidden ? null : (
                <>
                  <Row>
                    <Column>Resultater</Column>
                  </Row>
                  {this.reviews.map((review, index) => (
                    <Row key={index}>
                      <ReviewCard
                        title={review.review_title}
                        subtitle={review.game_title}
                        terningkast={review.rating}
                        relevanse={review.likes}
                        text={review.text}
                      >
                        <Button.Success
                          small
                          onClick={() => {
                            history.push('/publishedReviews/' + review.review_id);
                          }}
                        >
                          Les mer
                        </Button.Success>
                      </ReviewCard>
                    </Row>
                  ))}
                </>
              )}
            </ColumnCentre>
          </Row>
        </Container>
      </>
    );
  }

  mounted() {
    reviewService
      .getPublishedReviews()
      .then((reviews) => (this.reviews = reviews))
      .catch((error) => Alert.danger('Error getting reviews: ' + error.message));
  }
}

/**
 * 
export class Platform extends Component {
  platforms: Platform[] = [];
  genres: Genre[] = [];
  reviews: Review[] = [];

  review: Review = {
    review_id: 0,
    game_id: 0,
    game_title: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
    genre_id: 0,
    relevant: 0,
    platform_id: 0,
    likes: 0,
  };
  state = { isHidden: false };
  platformCall(id: number) {
    reviewService
      .getPlatform(id)
      .then((data) => {
        //this.setState({ isHidden: false });
        console.log(data);
        this.reviews = data;
      })
      .catch((error) => Alert.danger('Error retrieving reviews: ' + error.message));
  }
  render() {
    return (
      <>
        <Container>
          <h3 className="text-center">Her kan du søke anmeldelser etter platform</h3>
          <div className="text-center">
            <Button.Success
              small
              onClick={() => {
                history.push('/reviews-by-genre');
              }}
            >
              Klikk her
            </Button.Success>{' '}
            for å søke etter sjanger
          </div>
          <Linebreak></Linebreak>
          <h5>Velg platform:</h5>

          <Row>
            <Column width={1}>
              <Card title="Play Station 4">
                <Button.Success onClick={() => this.platformCall(145)}>Open</Button.Success>
              </Card>
            </Column>
            <ColumnCentre width={6} smwidth={4} mdwidth={2}>
              {this.genres.map((genre) => (
                <CategoryCard
                  key={genre.genre_id}
                  img={
                    genre.genre_img
                      ? 'https://helenaagustsson.github.io/INFT2002-images/images/' +
                        genre.genre_img +
                        '.png'
                      : 'https://helenaagustsson.github.io/INFT2002-images/images/' +
                        genre.genre_name +
                        '.png'
                  }
                >
                  <Button.Success
                    onClick={() => {
                      this.platformCall(genre.genre_id);
                    }}
                  >
                    {genre.genre_name}
                  </Button.Success>
                </CategoryCard>
              ))}
            </ColumnCentre>

            <ColumnCentre>
              {this.state.isHidden ? null : (
                <div className="sticky-top">
                  <div>Resultater:</div>
                  {this.reviews.map((review, index) => (
                    <Row key={index}>
                      <ReviewCard
                        title={review.review_title}
                        subtitle={review.game_title}
                        terningkast={review.rating}
                        relevanse={review.likes}
                        text={review.text}
                      >
                        <Button.Success
                          small
                          onClick={() => {
                            history.push('/publishedReviews/' + review.review_id);
                          }}
                        >
                          Les mer
                        </Button.Success>
                      </ReviewCard>
                    </Row>
                  ))}
                </div>
              )}
            </ColumnCentre>
          </Row>
        </Container>
      </>
    );
  }
  mounted() {
    reviewService
      .getPublishedReviews()
      .then((reviews) => (this.reviews = reviews))
      .catch((error) => Alert.danger('Error getting reviews: ' + error.message));
    //platformService.getAll().then((platforms)=>{this.platforms = platforms})
  }
}

 * 
 */
