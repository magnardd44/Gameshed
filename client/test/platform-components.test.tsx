import * as React from 'react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import { shallow } from 'enzyme';
import { history, Platform } from '../src/platform-components';
import { Review, reviewService } from '../src/services/review-service';

import {
  ReviewCard,
  ColumnCentre,
  Form,
  Card,
  Alert,
  Button,
  Container,
  Column,
} from '../src/widgets';

const mockAdapter = new MockAdapter(axios);

const completeReview = {
  review_id: 3,
  game_id: 1,
  game_title: 'abc',
  review_title: 'def',
  text: 'ghi',
  user_id: 1,
  rating: 4,
  published: true,
  genre_id: 1,
  platform_id: 145,
  relevant: 1,
  likes: 1,
  user_nickname: 'jkl',
};

const noReviews: Review[] = [];

const completeReviews = [
  {
    review_id: 1,
    game_id: 1,
    game_title: 'abc',
    review_title: 'def',
    text: 'ghi',
    user_id: 1,
    rating: 4,
    published: true,
    genre_id: 1,
    platform_id: 145,
    relevant: 1,
    likes: 1,
    user_nickname: 'jkl',
  },
  {
    review_id: 2,
    game_id: 2,
    game_title: 'abc',
    review_title: 'def',
    text: 'ghi',
    user_id: 2,
    rating: 2,
    published: true,
    genre_id: 2,
    platform_id: 145,
    relevant: 1,
    likes: 2,
    user_nickname: 'jkl',
  },
];

describe('Platform component', () => {
  test('Platform component default', () => {
    mockAdapter.onGet('/platforms/').reply(200, []);

    const wrapper = shallow(<Platform />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Button go to genre', () => {
    const wrapper = shallow(<Platform />);

    const spy = jest.spyOn(history, 'push');

    wrapper.find({ children: 'Klikk her' }).simulate('click');

    expect(spy).toBeCalledWith('/reviews-by-genre');
  });

  test('Test all platform buttons', () => {
    const wrapper = shallow(<Platform />);

    const spy = jest.spyOn(reviewService, 'getPlatform');

    wrapper.find({ children: 'Vis' }).forEach((b) => b.simulate('click'));

    expect(spy).toBeCalledTimes(11);
  });

  test('Got no reviews', (done) => {
    const wrapper = shallow(<Platform />);

    mockAdapter.onGet('/reviews/platform/145').reply(200, noReviews);

    wrapper.find({ children: 'Vis' }).at(0).simulate('click');

    setTimeout(() => {
      expect(reviewService.reviews).toHaveLength(0);
      done();
    });
  });

  test('Test reviews are shown', (done) => {
    const wrapper = shallow(<Platform />);

    ///reviewService.reviews = completeReviews

    setTimeout(() => {
      //expect(wrapper.find('Les mer')).toHaveLength(2)
      done();
    });
    //expect(spy).toBeCalledTimes(11);
  });

  //  test('Test getting reviews by platform', (done) => {
  //    const wrapper = shallow(<Platform />);
  //
  //	mockAdapter.onGet('/reviews/platform/145').reply(200, completeReviews);
  //	mockAdapter.onGet('/reviews/review/1').reply(200);
  //	mockAdapter.onGet('/reviews/review/2').reply(200);
  //
  //    //const spy = jest.spyOn(reviewService, 'getPlatform');
  //
  //    wrapper.find({ children: 'Vis' }).at(0).simulate('click');
  //
  //	setTimeout(()=>{
  //		expect(reviewService.reviews[0].review_id).toBe(1)
  //		expect(reviewService.reviews[1].review_id).toBe(2)
  //	});
  //    //expect(spy).toBeCalledTimes(11);
  //  });
});
