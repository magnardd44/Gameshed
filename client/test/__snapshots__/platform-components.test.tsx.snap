// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platform component Platform component default 1`] = `
<Fragment>
  <Container>
    <h3
      className="text-center"
    >
      Her kan du søke anmeldelser etter platform
    </h3>
    <div
      className="text-center"
    >
      <ButtonSuccess
        onClick={[Function]}
        small={true}
      >
        Klikk her
      </ButtonSuccess>
       
      for å søke etter sjanger
    </div>
    <Linebreak />
    <h5>
      Velg platform:
    </h5>
    <Row>
      <ColumnCentre
        mdwidth={2}
        smwidth={4}
        width={6}
      >
        <Card
          title="Play Station 4"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
        <Card
          title="Play Station 5"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
        <Card
          title="Nintendo Switch"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
        <Card
          title="XBox One"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
        <Card
          title="XBox 360"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
        <Card
          title="PC - Microsoft"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
        <Card
          title="Gameboy"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
        <Card
          title="Mac"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
        <Card
          title="IOS"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
        <Card
          title="Nintendo DS"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
        <Card
          title="Andre"
        >
          <ButtonSuccess
            onClick={[Function]}
          >
            Vis
          </ButtonSuccess>
        </Card>
      </ColumnCentre>
      <ColumnCentre>
         
         
      </ColumnCentre>
    </Row>
  </Container>
</Fragment>
`;
